wp/lib/bap_wp/src/precondition.ml

```diff
@@ -1307,7 +1330,7 @@ let user_func_spec
           match r with
           | Some q -> q
           | None -> let q, _ = Env.mk_init_var env v in q) in
-      let tid_name : string = Tid.name tid in
+      let tid_name : string = Tid.create () |> Tid.to_string in
```

Comment:
Why do we create a new tid here?

Reply:
The old behavior was a bug in the case where a summarized function is called multiple times.  It resulted us generating the same name for the return memories of the different calls.